src/Data/Foreign.purs

```diff
@@ -137,3 +139,7 @@ readInt value = either (const error) fromNumber (readNumber value)
 readArray :: Foreign -> F (Array Foreign)
 readArray value | isArray value = pure $ unsafeFromForeign value
 readArray value = Left (TypeMismatch "array" (tagOf value))
+
+newtype Prop = Prop { key :: String, value :: Foreign }
```

Comment:
It might be better to use a non-record type here, as data constructors with multiple fields perform better than records in most JS VMs (cheaper to construct due to the "hidden class" optimisation). Maybe just a `type Prop = Tuple String Foreign` even?